core/src/smartcontracts/isi/account.rs

```diff
@@ -29,13 +29,19 @@ pub mod isi {
             wsv: &WorldStateView,
         ) -> Result<(), Self::Error> {
             let asset_id = self.object.id();
+            let account_id = asset_id.account_id.clone();
 
             match wsv.asset(asset_id) {
                 Err(err) => match err {
                     QueryExecutionFailure::Find(find_err)
                         if matches!(*find_err, FindError::Asset(_)) =>
                     {
                         assert_can_register(&asset_id.definition_id, wsv, self.object.value())?;
+                        let latest_tx_hash = wsv.latest_transaction_hash();
```

Comment:
Hm... So you use not the hash of current block, but the hash of the previous one. However docs for `Account::add_asset_register_tx()` and the corresponding field don't document this